src/test/java/org/jooq/lambda/IteratorsTest.java

```diff
@@ -0,0 +1,91 @@
+package org.jooq.lambda;
+
+import org.junit.Test;
+
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+import java.util.function.Predicate;
+
+import static org.junit.Assert.*;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * Created by billoneil on 7/26/17.
+ */
+public class IteratorsTest {
+
+    @Test(expected = NoSuchElementException.class)
+    public void testPeekingThrows() {
+        Seq<Integer> seq1 = Seq.of();
+        PeekingIterator<Integer> peeking1 = new Iterators.PeekingIteratorImpl<>(seq1.iterator());
+        peeking1.next();
+    }
+
+    @Test
+    public void testPeeking() {
+        Seq<Integer> seq1 = Seq.of();
+        PeekingIterator<Integer> peeking1 = new Iterators.PeekingIteratorImpl<>(seq1.iterator());
+        assertFalse(peeking1.hasNext());
+
+        Seq<Integer> seq2 = Seq.of(1, 2, 3);
+        PeekingIterator<Integer> peeking2 = new Iterators.PeekingIteratorImpl<>(seq2.iterator());
+        int i = 0;
+        while (peeking2.hasNext()) {
+            assertEquals(++i, (int) peeking2.next());
+        }
+
+        Seq<Integer> seq3 = Seq.of(1, 2, 3);
+        PeekingIterator<Integer> peeking3 = new Iterators.PeekingIteratorImpl<>(seq3.iterator());
+        assertEquals(1, (int) peeking3.peek());
+        assertEquals(1, (int) peeking3.peek());
+        assertEquals(1, (int) peeking3.next());
+        assertEquals(2, (int) peeking3.next());
+        assertEquals(3, (int) peeking3.peek());
+        assertEquals(3, (int) peeking3.peek());
+        assertEquals(3, (int) peeking3.next());
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testCountingPredicateThrowsIllegalArg() {
```

Comment:
I think they are in the same package so it should have been package-private. The first implementation was must have missed it when I cleaned it up.

>2. I'm not too fan of testing internals with unit tests. The internals will be tested transitively by testing the chunked API. With tests on internals, refactoring those internals will be much harder...

👍 totally agree wasn't sure of your preference so opted for more testing